src/trg-client.c

```diff
@@ -325,7 +340,15 @@ gchar *trg_client_get_url(TrgClient * tc)
 gchar *trg_client_get_session_id(TrgClient * tc)
 {
     TrgClientPrivate *priv = tc->priv;
-    return priv->session_id ? g_strdup(priv->session_id) : NULL;
+    gchar *ret;
+
+    g_mutex_lock(&priv->configMutex);
```

Comment:
This could use some explanation, perhaps in a separate commit.

Reply:
I just saw that writing to priv->session_id was protected by the mutex so I though non-atomic reading it should be too. I haven't had any issues with this code, so maybe it's not possible that a race condition can happen in practice, but IMHO it's better to be safe than sorry when it comes to threads

Reply:
Could you just put it in its own commit with a message similar to what you said here.

Reply:
Ok, will do!